src/ui/common/src/reducers/workflow.ts

```diff
@@ -464,8 +448,82 @@ export const handleGetSelectDagPosition = createAsyncThunk<
     const dag = thunkAPI.getState().workflowReducer.selectedDag;
     const artifactResults =
       thunkAPI.getState().workflowReducer.artifactResults ?? {};
+
     if (!!dag) {
-      return collapsePosition(allNodes, dag, artifactResults);
+      const collapsedPosition = collapsePosition(
+        allNodes,
+        dag,
+        artifactResults
+      );
+
+      const elk = new ELK();
+      const mappedNodes = collapsedPosition.nodes.map((node) => {
+        return {
+          id: node.id,
+          // Width and height set not only the node width/height, but also how far apart we want to position these nodes.
+          // So using a value of just the width and the height of the node will result in no spacing in between.
+          // Elk (eclipse layout kernel) also provides many knobs to tweak this as well.
+          width: 300,
+          height: 300,
+        };
+      });
+
+      // TODO: move into collapsed nodes function.
```

Comment:
nit: add a linear task

Reply:
Done!
https://linear.app/aqueducthq/issue/ENG-2303/chore-move-logic-for-filtering-out-missing-edges-into-collapse-nodes